WSUCptSCapstone-Fall2022Spring2023/psd-gamifiedapp
# Sprint4/visuals

closes #49 

Demo Video Link: https://youtu.be/JuLbqAJodBk

Overview:

This PR covers the addition of finished polished visuals for the first 2 main levels of the game, titled "Life Support" and "Shield Generators". Visual includes the addition of background art for each phase. Dynamics within that art, and more polished UI elements.

Details:

InitBarcode: Backing code for the "Barcode" widget
WatchString: Backing code for the UI string display
ProportionalMeterHandler: Backing code for the proportional meter UI element.
MeterHandler: Backing code for the numeric meter element.
WatchNumber: Backing code for the numeric display element.
NumberLineHandler: Backing code for the number line UI element.
WatchBoolean: Backing code for the boolean display element.
PhaseUIController: Changes were made here to allow action groups to pass events through to phase UI
LevelPlayerController: Changes were made to allow for visuals to reset on failure.
PhaseDefinition: Added reset groups to the action group structure.
IronPythonContainer.cs: Added additional bindings to invoke reset commands.
DynamicRotate.cs: Rotates a UI element at a variable rate when enabled.
DynamicProgression: When enabled, transitions through a progression of multiple images.
DynamicFlickerTransition: When enabled, flicker from one image to another.
DynamicColorFlicker: When enabled, flicker an element from it's current color to another color.

Testing:

Due to the mostly visual nature of this PR, the following was our manual testing procedure.

1. Start Game
2. Play through levels 1 & 2 entirely
3. In each phase:
4. Ensure that all expected visual reactions trigger
5. Ensure that all UI elements respond to backing variables as expected.


In the current build the layouts positions and behaviors pass this process as expected.
